src/script/interpreter.cpp

```diff
@@ -1432,6 +1432,46 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                 }
                 break;
 
+                case OP_SHA256_INIT: // (in -- sha256_ctx)
```

Comment:
I think separating processing of these two opcodes in different code blocks might be beneficial: it will improve the readability and lessen the number of branches in the code, while not significantly increasing the size of the code (cache impact)

Reply:
I agree.  Combining them turned out way worse than I was anticipating.